### PR TITLE
[webui] Add url to project meta

### PR DESCRIPTION
**Pull request by dwmh**
Display it on project page as link.

Co-Author: Esa Kulmala <esa.kulmala@ericsson.com>

### CODE REVIEW

File: src/api/app/views/webui/project/show.html.erb [hunk 1]
```diff
@@ -11,6 +11,11 @@
   <div class="grid_10 alpha">
     <h3 id="project_title"><%= @project.title %><%= @project if @project.title.blank? -%></h3>
     <%= description_wrapper(@project.description) %>
+    <% unless @project.url.blank? %>
+      <div style="margin: 10px" >
```
Comment (ChrisBr) [inline src/api/app/views/webui/project/show.html.erb]: Please do not use inline styles!

File: src/api/app/views/webui/project/show.html.erb [hunk 2]
```diff
@@ -11,6 +11,11 @@
   <div class="grid_10 alpha">
     <h3 id="project_title"><%= @project.title %><%= @project if @project.title.blank? -%></h3>
     <%= description_wrapper(@project.description) %>
+    <% unless @project.url.blank? %>
```
Comment (ChrisBr) [inline src/api/app/views/webui/project/show.html.erb]: Please use ``present?`` instead
http://apidock.com/rails/Object/present%3F

File: src/api/app/views/models/_project.xml.builder [hunk 3]
```diff
@@ -14,6 +14,7 @@ xml.project(project_attributes) do
     end
   end
 
+  xml.url(my_model.url) unless my_model.url.blank?
```
Comment (ChrisBr) [inline src/api/app/views/models/_project.xml.builder]: Please use ``present?`` instead 
http://apidock.com/rails/Object/present%3F
(However, would be fine in sticking to ``blank?`` as the following lines also use ``blank?``)

File: src/api/app/views/webui/project/show.html.erb [hunk 4]
```diff
@@ -11,6 +11,11 @@
...
+      <div style="margin: 10px" >
+        <%= link_to(@project.url, @project.url) %>
```
Comment (ChrisBr) [inline src/api/app/views/webui/project/show.html.erb]: Maybe also worth to discuss to make the ``<h3>`` element a link?

**Comment by ChrisBr** [comment]
@dwmh thanks for your contribution. Except the failing tests it LGTM :+1:

**Comment by dwmh** [comment]
Fair enough, I'll make the changes probably later today.

**Comment by adrianschroeter** [comment]
This can break interconnect to older instances when used. We need to verify if we need backports.

Given that it is not used for building, an attribute might be an alternative solution for this ...

**Comment by hennevogel** [comment]
Just a general question, what is this url supposed to be? And you do know that this already works for packages right? :-)

**Comment by dwmh** [comment]
Typically we use this field to point to an outside CI job that generated the project in OBS. It's also useful when the project has some web page associated with it. The package level url was not enough for those purposes.

I think attribute could also be used, but this is the solution we have deployed internally.

**Comment by hennevogel** [comment]
So either we backport this to 2.7 or we filter this in the interconnect...

**Comment by adrianschroeter** [comment]
We accept this now also in 2.7. It would only break there if someone is using the feature, so we can merge it now.

But the pull request needs to be rebased due to the conflicts in structure.sql

2.7 commit:
370135078b8d130f57d80c7fd595be098117e984

**Pull request merged**